.github/workflows/snapsync-test.yaml

```diff
@@ -0,0 +1,194 @@
+
+        echo "NETWORK=$NETWORK"
+        echo "NODE_REF=$NODE_REF"
+        echo "GETH_REF=$GETH_REF"
+        echo "USE_DEV_IMAGE=$USE_DEV_IMAGE"
+
+        echo "NETWORK=$NETWORK" >> $GITHUB_ENV
+        echo "NODE_REF=$NODE_REF" >> $GITHUB_ENV
+        echo "GETH_REF=$GETH_REF" >> $GITHUB_ENV
+        echo "USE_DEV_IMAGE=$USE_DEV_IMAGE" >> $GITHUB_ENV
+
+    # despite no repo code being used, need checkout in order to create a shared workspace between steps
+    - name: Checkout
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Install pre-requisites
+      id: prereq
+      shell: bash
+      run: |-
+        # Update and upgrade packages
+        sudo apt-get update
+        sudo apt-get upgrade -y
+
+        # Install pre-requisites
+        sudo apt-get install -y curl bc
+
+        ### yq to manipulate yaml
+        BINARY=yq_linux_amd64
+        VERSION=v4.45.1
+
+        wget https://github.com/mikefarah/yq/releases/download/${VERSION}/${BINARY}.tar.gz -O - | tar xz && sudo mv ${BINARY} /usr/bin/yq
+
+    - name: Verify Docker and docker-compose installation
+      id: docker-verify
+      shell: bash
+      run: |-
+        docker run hello-world
+        docker compose version
+
+    - name: Checkout celo-l2-node-docker-compose repo
+      uses: actions/checkout@v4
+      with:
+        repository: celo-org/celo-l2-node-docker-compose
+        path: celo-l2-node-docker-compose
+        fetch-depth: 0
+
+    - name: Configure l2 node project
+      id: configure-l2-node
+      shell: bash
+      run: |-
+        cd celo-l2-node-docker-compose
+
+        # copy network specific env file
+        cp ${{ env.NETWORK }}.env .env
+
+        # transform volume paths to be compatible with parent container directory (for self hosted github runners)
+        export RUNNER_PWD_PATH=$(echo "$(pwd)" | sed 's|^/__w/|/runner/_work/|')
+
+        yq eval ".services[].volumes |= map(sub(\"^\\./\"; env(RUNNER_PWD_PATH) + \"/\"))" -i docker-compose.yml
+
+        # interpolate image refs into docker-compose.yml
+        op_node="us-west1-docker.pkg.dev/devopsre/celo-blockchain-public/op-node:${{ env.NODE_REF }}"
+        op_geth="us-west1-docker.pkg.dev/devopsre/celo-blockchain-public/op-geth:${{ env.GETH_REF }}"
+
+        # use dev image if specified
+        if [[ "${{ env.USE_DEV_IMAGE }}" == "true" ]]; then
+          op_node="us-west1-docker.pkg.dev/blockchaintestsglobaltestnet/dev-images/op-node:${{ env.NODE_REF }}"
+          op_geth="us-west1-docker.pkg.dev/blockchaintestsglobaltestnet/dev-images/op-geth:${{ env.GETH_REF }}"
+        fi
+
+        yq eval ".services.op-node.image = \"$op_node\"" docker-compose.yml -i
+        yq eval ".services.op-geth.image = \"$op_geth\"" docker-compose.yml -i
+
+    - name: Start l2 node
+      id: start-l2-node
+      shell: bash
+      run: |-
+        cd celo-l2-node-docker-compose
+
+        docker-compose up --build -d
```

Comment:
I think it would be good to save the op-geth log output and make it downloadable as a CI artifact in case the test fails.